be/src/exprs/fpe.cpp

```diff
@@ -0,0 +1,206 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <cmath>
+#include <vector>
+#include <ubiq/fpe/ff1.h>
+#include <iomanip>
+
+#include "exprs/fpe.h"
+#include "util/defer_op.h"
+
+namespace starrocks {
+
+const std::string_view FPE::DEFAULT_KEY = "abcdefghijk12345abcdefghijk12345";
+
+std::string_view FPE::trim_leading_zeros(const std::string_view& str) {
+    size_t start = str.find_first_not_of('0');
+    if (start == std::string::npos) {
+        return "0";
+    }
+    return str.substr(start);
+}
+
+std::string_view FPE::trim_trailing_zeros(const std::string_view& str) {
+    size_t end = str.find_last_not_of('0');
+    if (end == std::string::npos) {
+        return str;
+    }
+    return str.substr(0, end + 1);
+}
+
+Status FPE::encrypt(const std::string_view& num_str, const std::string_view& key, char* buffer, size_t* len, int radix = 10) {
```

Comment:
There are many string copies in this method introduced by `string::sub_str`, `string::operator +`, and `string::operator =`.
Could we just apply all (or some) char actions on a `char* buffer` to avoid these copies?